### ISSUE TITLE
Any script with a `class_name` starting with `Editor` (case-sensitive) does not appear in Add Node dialog

### ISSUE THREAD

**Issue by Yagich** [godotengine/godot#68318]
### Godot version

4.0b4

### System information

Linux/Manjaro

### Issue description

Pretty much as it says in the title. I was not able to find anything documenting this behavior, so at least documenting it would be useful.

### Steps to reproduce

1. Make a Node-extending script and set its `class_name` to something beginning with `Editor`;
2. Try to add your new Node to a scene by filtering for it in the Add Node dialog.

**Comment by AThousandShips** [comment]
This is by design, not a bug, to prevent the built in nodes starting with editor showing up, as they cannot be created in a project

This should maybe be documented somewhere though, or give an error in the script editor

**Comment by Yagich** [comment]
Yes, to be clear I don't think of it as a bug slightly_smiling_face I thought that creating a whole proposal to document what is a pretty tiny thing would be overkill, so I posted it as an issue here. Perhaps it would be a better fit as an issue in godot-docs?

**Comment by AThousandShips** [comment]
Wouldn't hurt to open one there too perhaps, not sure where this would fit best tbh in the documentation, adding a warning would be extensive I think not familiar with exactly where that is checked, will take a look later today perhaps

**Comment by MewPurPur** [comment]
Related to #56798

**Comment by theraot** [comment]
Bug? This is a feature :v - it is not exactly what I wanted, but it might do, see https://github.com/godotengine/godot-proposals/issues/9298
